tools/cmock/coverage.cmake

```diff
@@ -69,7 +69,7 @@ execute_process(COMMAND lcov
 # remove source files from dependencies and unit tests
 execute_process(COMMAND lcov
     --rc lcov_branch_coverage=1
-    --remove ${CMAKE_BINARY_DIR}/coverage.info *dependency* *unit-test* /usr* */source/ota.c *CMakeCCompilerId*
+    --remove ${CMAKE_BINARY_DIR}/coverage.info *dependency* *unit-test* /usr* */source/ota.c *CMakeCCompilerId* */source/portable/os/ota_os_posix.c
```

Comment:
Why are we removing `ota.c` and `ota_os_posix.c` from coverage checks?

Reply:
For ota.c, we make a copy of it with the static functions stripped off and use that in the unit test, hence we don’t use the source version of the file for the coverage report.
ota_os_posix.c provides abstractions for Linux and not very critical to the coverage report of the library. We aren’t removing these altogether and still have unit tests that assert the basic functionality of these abstractions. The user may or may not use the exact implementation in their code.